Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -337,16 +383,38 @@ public:
                     index_type const cell_stop_1  = cell_offsets_1[i_cell+1];
                     index_type const cell_half_1 = (cell_start_1+cell_stop_1)/2;
 
-                    // Same but for the pairs
-                    index_type const cell_start_pair = have_product_species?
-                                                           p_pair_offsets[i_cell] : 0;
-
                     // Do not collide if there is only one particle in the cell
                     if ( cell_stop_1 - cell_start_1 <= 1 ) { return; }
 
                     // shuffle
                     ShuffleFisherYates(
                         indices_1, cell_start_1, cell_half_1, engine );
+                }
+            );
+
+            // Loop over cells
+            amrex::ParallelForRNG( n_independent_pairs,
+                [=] AMREX_GPU_DEVICE (int i_coll, amrex::RandomEngine const& engine) noexcept
+                {
+                    // to avoid type mismatch errors
+                    auto ui_coll = (index_type)i_coll;
+
+                    // Use a bisection algorithm to find the index of the cell in which this pair is located
+                    int i_cell = amrex::bisect( p_coll_offsets, 0, n_cells, ui_coll );
```

Comment:
To fix `clang-tidy` error:
```suggestion
                    const int i_cell = amrex::bisect( p_coll_offsets, 0, n_cells, ui_coll );
```